libraries/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -139,13 +156,17 @@ export class ProjectWatcher {
 
         this._hasRenderedStatus = false;
 
-        const resolveIfChanged = async (): Promise<void> => {
+        const resolveIfChanged: () => Promise<void> = (this._resolveIfChanged = async (): Promise<void> => {
```

Comment:
If this feels messy, feel free to assign `this._resolveIfChanged = resolveIfChanged` on the next line instead.